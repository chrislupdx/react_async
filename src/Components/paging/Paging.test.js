import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

describe('Paging component', () => {
  it('renders Paging', () => {
    const wrapper = shallow(
      <Paging currentPage={1} totalPages={10} nextPage={jest.fn()}>
        <h1>hi</h1>
      </Paging>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('increments page on next button click', () => {
    const next = jest.fn();
    const wrapper = shallow(
      <Paging currentPage={2} totalPages={10} nextPage={next}>
        <h1>heyy</h1>
      </Paging>
    );

    wrapper.find('button').at(1).simulate('click');

    expect(next).toHaveBeenCalledWith(3);
  });

  it('decrements page on previous button click', () => {
    const next = jest.fn();
    const wrapper = shallow(
      <Paging currentPage={2} totalPages={10} nextPage={next}>
        <h1>hi</h1>
      </Paging>
    );
    wrapper.find('button').at(0).simulate('click');
    expect(next).toHaveBeenCalledWith(1);
  });

  it('does not display previous button on page 1', () => {
    const next = jest.fn();
    const wrapper = shallow(
      <Paging currentPage={1} totalPages={10} nextPage={next}>
        <h1>hi</h1>
      </Paging>
    );

    const buttons = wrapper.find('button');

    expect(buttons).toHaveLength(1);
    expect(buttons.at(0).text()).toEqual('Next Page');
  });

  it('does not display the next button when on last page', () => {
    const next = jest.fn();
    const wrapper = shallow(
      <Paging currentPage={10} totalPages={10} nextPage={next}>
        <h1>hi</h1>
      </Paging>
    );

    const buttons = wrapper.find('button');

    expect(buttons).toHaveLength(1);
    expect(buttons.at(0).text()).toEqual('Previous Page');
  });

    
})
;
