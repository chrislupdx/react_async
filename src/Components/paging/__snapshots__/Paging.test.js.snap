// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders Paging 1`] = `
<Fragment>
  1 / 10
  <button
    onClick={[Function]}
  >
    Next Page
  </button>
  <h1>
    hi
  </h1>
</Fragment>
`;
